src/app/services/student-round.service.ts

```diff
@@ -145,6 +145,45 @@ export class StudentRoundService {
     shareReplay(1),
   );
 
+  interactions$: Observable<Interaction[]> =
+  combineLatest([this.sessionService.currentRoundPath$, this.authService.currentUser$]).pipe(
+    switchMap(([path, user]) =>
+      path && user
+        ? this.firebaseService.getStudentInteractions(path, user.uid)
+        : of([])),
+    distinctUntilChanged()
+  );
+
+  totalPollen$: Observable<Interaction[]> = this.interactions$.pipe(
+    map(interactions =>
+      interactions.filter(interaction => interaction.barcodeValue >= 1 && interaction.barcodeValue <= 16)
+    ),
+    distinctUntilChanged(),
```

Comment:
Again, is `distinctUntilChanged()` really doing anything here for this array? If this observable output just a number then it would work as is.